analyses/cell-type-ewings/cell_type_annotations/SCPCS000492/SCPCL000824_tumor-gene-expression-exploration.Rmd

```diff
@@ -0,0 +1,670 @@
+We also see that `AUCell` identifies a threshold based on the local minima of the AUC distribution where all cells above the threshold are classified as expressing the gene set.
+Let's compare the assignments from our validated classifications to using `AUCell` with `SCPCL000822`. 
+
+```{r}
+# compare AUCell classification vs cnv classification in ref
+# pull out tumor cells found by AUCell
+tumor_cells <- ref_assignment$markers$assignment
+
+ref_classification_df <- ref_classification_df |> 
+  # add new column with auc classification 
+  dplyr::mutate(auc_classification = dplyr::if_else(barcodes %in% tumor_cells, "Tumor", "Normal"))
+
+# filter out any ambiguous cells 
+caret_df <- ref_classification_df |> 
+  dplyr::filter(tumor_cell_classification != "Ambiguous")
+
+# compare using a confusion matrix
+caret::confusionMatrix(
+    table(
+      caret_df$tumor_cell_classification, 
+      caret_df$auc_classification)
+  ) 
+```
+
+It looks like we get pretty good agreement between our original classification and the updated `AUCell` classification using the marker gene list.
+
+Now we repeat on `SCPCL000824`. 
+
+```{r}
+# get new auc and plot thresholds 
+new_auc <- AUCell::AUCell_run(counts(sce), collection)
+new_assignment <- AUCell::AUCell_exploreThresholds(new_auc, assign = TRUE)
+```
+
+
+Here we do not see any bimodal distributions and that most cells are considered to not be expressing the specified gene sets, which doesn't fit with our previous findings. 
+I expect that without a bimodal distribution this model does not do a good job of calculating a threshold. 
+
+Instead, we will use the threshold identified by `AUCell` for `SCPCL000822` to identify tumor cells in `SCPCL000824`.  
+Here, we label any cells that have passed the threshold set for the marker gene sets as tumor cells. 
+
+```{r}
+# pull out threshold used for assigning cells from ref sample 
+ref_threshold <- ref_assignment$markers$aucThr$selected
+
+# create a vector of tumor cells in 824 using the cutoff from 822
+new_tumor_cells <- new_auc@assays@data$AUC |> 
+  as.data.frame() |> 
+  tibble::rownames_to_column("gene_set") |> 
+  tidyr::pivot_longer(!gene_set, 
+                      names_to = "barcodes", 
+                      values_to = "auc") |> 
+  dplyr::filter(gene_set == "markers", auc >= ref_threshold) |>
+  dplyr::pull(barcodes)
+
+# add to original classification df 
+classification_df <- classification_df |> 
+  dplyr::mutate(auc_classification = dplyr::if_else(barcodes %in% new_tumor_cells, "Tumor", "Normal"))
+```
+
+```{r}
+# visualize which cells are classified as tumor 
+ggplot(classification_df, aes(x = UMAP1, y = UMAP2, color = auc_classification)) +
+  geom_point(alpha = 0.5, size = 0.5)
+```
+
+This looks similar to using the updated marker gene cutoff to classify cells where most cells in that bottom group are tumor cells.  
```

Comment:
Just for completeness, do you want to do a confusion matrix for this AUC vs. previous version?